modules/node-shared/src/main/scala/io/constellationnetwork/node/shared/infrastructure/snapshot/GlobalSnapshotContextFunctions.scala

```diff
@@ -34,8 +34,98 @@ import eu.timepit.refined.types.all.NonNegLong
 abstract class GlobalSnapshotContextFunctions[F[_]] extends SnapshotContextFunctions[F, GlobalIncrementalSnapshot, GlobalSnapshotInfo]
 
 object GlobalSnapshotContextFunctions {
-  def make[F[_]: Async: Parallel: HasherSelector](snapshotAcceptanceManager: GlobalSnapshotAcceptanceManager[F]) =
+  def make[F[_]: Async: Parallel: HasherSelector: SecurityProvider](
+    snapshotAcceptanceManager: GlobalSnapshotAcceptanceManager[F],
+    updateDelegatedStakeAcceptanceManager: UpdateDelegatedStakeAcceptanceManager[F],
+    withdrawalTimeLimit: EpochProgress,
+    tessellation3MigrationStartingOrdinal: SnapshotOrdinal
+  ) =
     new GlobalSnapshotContextFunctions[F] {
+
+      private def acceptDelegatedStakes(
```

Comment:
we should put a todo so that we can not repeat the function here and in `GlobalSnapshotConsensusFunctions`